benchmarking/benchmarks/benchmarks.py

```diff
@@ -81,12 +83,18 @@ def _verifyBenchmark(self, benchmark, filename, is_post):
                 "one benchmark."
 
         for test in tests:
+            assert "metric" in test, "Metric field is missing in " + \
+                "benchmark {}".format(filename)
+
+            # no check is needed if the metric is generic
+            if test["metric"] == "generic":
+                return
```

Comment:
We may either check only one test exists for generic metric or check all tests in a json file to be generic metric.